pkg/types/builder.go

```diff
@@ -96,27 +99,58 @@ func (g *Builder) buildResource(res *schema.Resource, refs config.References, na
 		if comment.TerrajetOptions.FieldJSONTag != nil {
 			jsonTag = *comment.TerrajetOptions.FieldJSONTag
 		}
-		fieldType, err := g.buildSchema(sch, refs, append(names, fieldName.Camel))
+
+		tfPaths := append(tfPath, fieldName.Snake)
+		xpPaths := append(xpPath, fieldName.LowerCamel)
+
+		fieldType, err := g.buildSchema(sch, cfg, tfPaths, xpPaths, append(names, fieldName.Camel))
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "cannot infer type from schema of field %s", fieldName.Snake)
 		}
 
-		// Build the field path based on the fact that names is a slice of
-		// strings which contains the name of the parent fields sequentially.
-		// We skip the first element since it is the main CRD type.
-		fp := strings.Join(append(names[1:], fieldName.Camel), ".")
-		if ref, ok := refs[fp]; ok {
+		tfFieldPath := fieldPath(tfPaths)
+		xpFieldPath := fieldPath(xpPaths)
+		if ref, ok := cfg.References[tfFieldPath]; ok {
 			comment.Reference = ref
 			sch.Optional = true
 		}
-		field := types.NewField(token.NoPos, g.Package, fieldName.Camel, fieldType, false)
+
+		fieldNameCamel := fieldName.Camel
+		if e, ix := containsAt(cfg.Sensitive.CustomFieldPaths, tfFieldPath); e || sch.Sensitive {
+			if e {
+				cfg.Sensitive.CustomFieldPaths = remove(cfg.Sensitive.CustomFieldPaths, ix)
```

Comment:
This was something I considered but I didn't like that users would need to provide a map in that case and the value part of the map (xp/json path) is even unknown before the code generation. We could expect them to set some placeholder (or empty?) values but that sounded hacky. 
But anyway, as we discussed today, I will need to revisit getting custom configuration since we would need the distinction of `include into connection details but not sensitive` type of fields.